### PR TITLE
feat(reporter): Add id of current ORT run as label to ORT results

### PR DESCRIPTION
**Pull request by wkl3nk**
Add the id of the current ORT run as label to the ORT result in order to have it available during report and notification generation.

### CODE REVIEW

File: workers/notifier/src/main/kotlin/notifier/NotifierOrtResultGenerator.kt
Comment (oheger-bosch) [inline workers/notifier/src/main/kotlin/notifier/NotifierOrtResultGenerator.kt:69]: The class comment contains a description of the labels available to the notifier script. Please add documentation for this new label there as well.

File: workers/notifier/src/main/kotlin/notifier/NotifierOrtResultGenerator.kt
Comment (oheger-bosch) [inline workers/notifier/src/main/kotlin/notifier/NotifierOrtResultGenerator.kt:147]: Better name this `getOrtRunLabels()` to align with the other functions.

File: workers/notifier/src/test/kotlin/notifier/NotifierOrtResultGeneratorTest.kt
Comment (oheger-bosch) [inline workers/notifier/src/test/kotlin/notifier/NotifierOrtResultGeneratorTest.kt:73]: The value of the label should be checked, too.

File: workers/reporter/src/main/kotlin/reporter/ReporterWorker.kt
Comment (oheger-bosch) [inline workers/reporter/src/main/kotlin/reporter/ReporterWorker.kt:42]: While I have no objections to add this label here for the reporter, just wondering whether there is already a concrete use case for this?
Alternatively, it would be an option to make the `NotifierOrtResultGenerator` class more generic, move it to `:workers:common` and use it for both notifier and reporter. This would reduce code duplication and provide a uniform set of labels for both workers.

File: workers/notifier/src/main/kotlin/notifier/NotifierOrtResultGenerator.kt
Comment (wkl3nk) [reply workers/notifier/src/main/kotlin/notifier/NotifierOrtResultGenerator.kt]: Function no longer needed after latest refactoring of PR.

File: workers/reporter/src/main/kotlin/reporter/ReporterWorker.kt
Comment (wkl3nk) [reply workers/reporter/src/main/kotlin/reporter/ReporterWorker.kt]: There is a feature request about having the _runId_ as label in the static web report. Therefore, it needs to be included also in the OrtResult used by the _reporter_ worker.
Refactoring: In order to get rid of the code duplication and have a place where all **common labels** can be specified, I extracted the duplicated code into a util class, which also adds the **common labels** that are used both by the _reporter_ and the _notifier_ worker.

File: workers/common/src/main/kotlin/common/OrtBaseResultGenerator.kt
Comment (oheger-bosch) [inline workers/common/src/main/kotlin/common/OrtBaseResultGenerator.kt:36]: I like the general direction of this refactoring. Just one thought: Would it make sense - instead of introducing a new object - to add the `generate()` function directly to `OrtRunService`? The current implementation suffers from [feature envy](https://refactoring.guru/smells/feature-envy).

File: workers/common/src/main/kotlin/common/OrtBaseResultGenerator.kt
Comment (oheger-bosch) [inline workers/common/src/main/kotlin/common/OrtBaseResultGenerator.kt:39]: The function could support a map parameter for additional labels to be added which defaults to an empty map. This would simplify its usage for clients; they would then not have to bother with copying the result and adding further labels.

**Review by oheger-bosch — COMMENTED** [review]
As a general note: I would recommend to do the introduction of the new result generator class/function in a separate commit. Currently, the single commit does a lot of things.

File: workers/common/src/main/kotlin/common/OrtBaseResultGenerator.kt
Comment (wkl3nk) [reply workers/common/src/main/kotlin/common/OrtBaseResultGenerator.kt]: Yes, I had a look and it makes a lot of sense, as the OrtRunService was "injected" as a parameter and did a lot of things using the OrtRunService. So it is better to have all the code at the same place. I got rid of this OrtBaseResultGenerator and put it into OrtRunService.
Some of the test cases also had then to be moved to OrtRunServiceTest.

File: workers/common/src/main/kotlin/common/OrtBaseResultGenerator.kt
Comment (wkl3nk) [reply workers/common/src/main/kotlin/common/OrtBaseResultGenerator.kt]: Please let's have a separate feature request for that.

Comment (oheger-bosch) [reply workers/common/src/main/kotlin/common/OrtBaseResultGenerator.kt]: This was not meant as a new feature request, but as a means to simplify the current implementation - the result generator for the notifier would directly benefit from it. But it's fine with me, I won't insist.

File: workers/notifier/src/test/kotlin/notifier/NotifierOrtResultGeneratorTest.kt
Comment (oheger-bosch) [inline workers/notifier/src/test/kotlin/notifier/NotifierOrtResultGeneratorTest.kt:206]: Can the other `every` declarations now be dropped? I think, on the mock only the new `generateOrtResult()` function will be called.

File: workers/common/src/main/kotlin/common/OrtRunService.kt
Comment (oheger-bosch) [inline workers/common/src/main/kotlin/common/OrtRunService.kt:285]: Nit: In contrast to Java, the convention for Kotlin comments is to use an imperative style, i.e. "Load the results...". I am personally also more used to the Java conventions, but throughout the codebase the Kotlin style is used.

File: workers/notifier/src/test/kotlin/notifier/NotifierOrtResultGeneratorTest.kt
Comment (wkl3nk) [reply workers/notifier/src/test/kotlin/notifier/NotifierOrtResultGeneratorTest.kt:206]: I am not sure if I understand correct. The `OrtRunService` is completely mocked, and it is used by test cases to check if 3 types of notifier-specific _labels_ are added. As this `generateOrtResult` function is called every time in order to add the common _labels_, I **have** to mock it, else the test does not find the function on the OrtRunService.

Comment (oheger-bosch) [reply workers/notifier/src/test/kotlin/notifier/NotifierOrtResultGeneratorTest.kt:206]: The `generateOrtResult()` has to be mocked for sure. I meant the other stubbing declarations, e.g. for `getXXXJobForOrtRun()`. They were necessary before, as they used to be called by the result generator. But now the result generator only calls `generateOrtResult()` and no longer the other functions.

**Review by oheger-bosch — APPROVED** [review]

File: workers/common/src/main/kotlin/common/OrtRunService.kt
Comment (sschuberth) [inline workers/common/src/main/kotlin/common/OrtRunService.kt:93]: Nit: This should drop the explicit type.

**Comment by mnonnenmacher** [comment]
@oheger-bosch Please have extra attention to the ECA check before adding PRs to the merge queue, we currently cannot make it required (https://gitlab.eclipse.org/eclipsefdn/it/api/git-eca-rest-api/-/issues/161, https://github.com/eclipse-apoapsis/.eclipsefdn/pull/4).

@wkl3nk Please do not edit commits with the GitHub UI as this makes the ECA check fail: https://api.eclipse.org/git/eca/status/gh/eclipse-apoapsis/ort-server/200

**Comment by oheger-bosch** [comment]
> @oheger-bosch Please pay extra attention to the ECA check before adding PRs to the merge queue, we currently cannot make it required (https://gitlab.eclipse.org/eclipsefdn/it/api/git-eca-rest-api/-/issues/161, [eclipse-apoapsis/.eclipsefdn#4](https://github.com/eclipse-apoapsis/.eclipsefdn/pull/4)).
> 
> @wkl3nk Please do not edit commits with the GitHub UI as this makes the ECA check fail: https://api.eclipse.org/git/eca/status/gh/eclipse-apoapsis/ort-server/200

What does this mean? Does the commit have to be dropped again?

**Comment by mnonnenmacher** [comment]
> > @oheger-bosch Please pay extra attention to the ECA check before adding PRs to the merge queue, we currently cannot make it required (https://gitlab.eclipse.org/eclipsefdn/it/api/git-eca-rest-api/-/issues/161, [eclipse-apoapsis/.eclipsefdn#4](https://github.com/eclipse-apoapsis/.eclipsefdn/pull/4)).
> > @wkl3nk Please do not edit commits with the GitHub UI as this makes the ECA check fail: https://api.eclipse.org/git/eca/status/gh/eclipse-apoapsis/ort-server/200
> 
> What does this mean? Does the commit have to be dropped again?

No, I think as we know the commit was authered by @wkl3nk and only the wrong email was used it should be fine.
However, looking at the git log the committer was overwritten anyway by `60134199+oheger-bosch@users.noreply.github.com` as you added the commit to the merge queue.

**Pull request merged**
